### PR TITLE
[trino] Adding catalog support

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Adding Trino catalog support.
- Examples: Catalog 'C1' with databases 'd1' and 'd2', and Catalog 'C2' with databases 'd2' and 'd3'. For Trino v1, display as 'C1.d1', 'C1.d2', 'C2.d2', 'C2.d3'."

## How was this patch tested?

- Locally

### CODE REVIEW

File: desktop/libs/notebook/src/notebook/connectors/trino.py [hunk 1]
```diff
@@ -79,15 +79,11 @@ def create_session(self, lang=None, properties=None):
...
-    user = self.user.username
-
-    return ClientSession(user, catalog)
-
-
   @query_error_handler
   def execute(self, notebook, snippet):
+    database = snippet['database']
+    query_client = TrinoQuery(self.db, 'USE ' + database)
+    query_client.execute()
```
Comment (Harshg999) [inline desktop/libs/notebook/src/notebook/connectors/trino.py:86]: Why do we need to execute this for every query?

File: desktop/libs/notebook/src/notebook/connectors/trino.py [hunk 2]
```diff
@@ -274,11 +270,17 @@ def close_session(self, session):
...
+    hostname, port = self.get_hostname_and_port(self.api_url)
+    databases = []
 
-    query_client = TrinoQuery(self.db, 'SHOW SCHEMAS')
-    response = query_client.execute()
-    res = response.rows
-    databases = [item for sublist in res for item in sublist]
+    for catalog in catalogs:
+      trino_session = ClientSession(self.user.username, catalog)
+      trino_request = TrinoRequest(hostname, port, trino_session)
```
Comment (Harshg999) [inline desktop/libs/notebook/src/notebook/connectors/trino.py:279]: Question: Not possible to re-use same session?

File: desktop/libs/notebook/src/notebook/connectors/trino.py (hunk 1 above)
Comment (agl29) [reply desktop/libs/notebook/src/notebook/connectors/trino.py:86]: We must execute the 'USE DB' query before every query to set the database, similar to how it's done in Hive [Hiveserver2](https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/hiveserver2.py#L331). Currently, multiple statement execution is not implemented. Therefore, we need to execute the 'USE DB' query before each query. In the future, we can consider implementing it only before the first statement.

File: desktop/libs/notebook/src/notebook/connectors/trino.py (hunk 2 above)
Comment (agl29) [reply desktop/libs/notebook/src/notebook/connectors/trino.py:279]: No, via `SET SESSION` cmd we can only set the properties of a session or catalog properties not the name.

**Review by Harshg999 — APPROVED** [review]

**Review by wing2fly — APPROVED** [review]

**Pull request merged**